### ISSUE TITLE
Cryptography ImportError

### ISSUE THREAD

**Issue by AloneWaker** [pyinstaller/pyinstaller#6390]
Pyinstaller version: 4.7
Cryptography version: 36.0.0

hi, when i try to create exe with pyinstaller i get this error:
from cryptography.hazmat.bindings._rust import
ImportError: DLL load failed while importing _rust: The specified module could not be found.
when i go to this path i see the file _rust.pyd

when i try other Cryptography version i get error with _padding import.

**Comment by rokm** [comment]
> ImportError: DLL load failed while importing _rust: The specified module could not be found.

This usually means that one of dependent DLLs could not be loaded. But I can't seem to reproduce it here.

What python version are you using (and where was it installed from), and on what Windows version are you trying to run the program?

**Comment by AloneWaker** [comment]
> > ImportError: DLL load failed while importing _rust: The specified module could not be found.
> 
> This usually means that one of dependent DLLs could not be loaded. But I can't seem to reproduce it here.
> 
> What python version are you using (and where was it installed from), and on what Windows version are you trying to run the program?

python 3.8 installed from python.org, windows 10.

**Comment by rokm** [comment]
Hmm, in that case, can you try making a simple `onedir` frozen application with just

```python
import cryptography.hazmat.bindings._rust
print(cryptography.hazmat.bindings._rust)
```

and upload it somewhere so I can take a look at it?

**Comment by AloneWaker** [comment]
> Hmm, in that case, can you try making a simple `onedir` frozen application with just
> 
> ```python
> import cryptography.hazmat.bindings._rust
> print(cryptography.hazmat.bindings._rust)
> ```
> 
> and upload it somewhere so I can take a look at it?

https://drive.google.com/drive/folders/1SePb1efGtw1mVUDngp3GI10hu6KyXiM3?usp=sharing

**Comment by rokm** [comment]
Hmm, `python3.dll` did not get collected for some reason, and it is required by `_rust.pyd`. But looking at your `build\Analysis-00.toc` file, it looks like you are using WIndows App store version of python, not python.org one. Is that intentional? (I'll check if I can reproduce missing `python3.dll` with that...)

**Comment by rokm** [comment]
Yeah, we miss the `python3.dll` on Windows App Store python; presumably because it is launched via shim and so the actual executable and its DLLs are not automatically in the search path when we try to resolve DLLs...

**Comment by AloneWaker** [comment]
> Hmm, `python3.dll` did not get collected for some reason, and it is required by `_rust.pyd`. But looking at your `build\Analysis-00.toc` file, it looks like you are using WIndows App store version of python, not python.org one. Is that intentional? (I'll check if I can reproduce missing `python3.dll` with that...)

ohh this was a mistake i did not notice i have python installed from microsoft store too, thank you for help.!

**Comment by User9000** [comment]
Is there a fix o this problem? i just encountered it.

**Comment by rokm** [comment]
@User9000 This particular problem (caused by python3.dll not being collected when using MS app store python) is fixed in PyInstaller 4.8. If you have a problem, open a new issue and provide relevant details there.

**Comment by User9000** [comment]
@rokm is there a way i can manually fix it? I just downloaded python 3.10 from python.org, and i started to get this issue.

**Comment by bwoodsend** [comment]
You haven't told us what your problem is so your guess is as good as ours.
